router/functionHandler.go

```diff
@@ -154,23 +156,37 @@ func (roundTripper RetryingRoundTripper) RoundTrip(req *http.Request) (resp *htt
 			log.Printf("Calling getServiceForFunction for function: %s", roundTripper.funcHandler.function.Name)
 
 			// send a request to executor to specialize a new pod
-			service, err := roundTripper.funcHandler.executor.GetServiceForFunction(
+			resp, err := roundTripper.funcHandler.executor.GetServiceForFunction(
 				roundTripper.funcHandler.function)
+
 			if err != nil {
-				log.Printf("Err from GetServiceForFunction : %v", err)
+				log.Debugf("Err from GetServiceForFunction : %v", err)
 				// We might want a specific error code or header for fission failures as opposed to
 				// user function bugs.
 				return nil, err
 			}
 
+			if resp.StatusCode != http.StatusOK {
+				if roundTripper.funcHandler.envType != "production" && len(req.Header.Get(fission.FissionDebugHeader)) != 0 {
+					return resp, nil
```

Comment:
So this is just the executor error, right?  Can we return something more useful to the user, like say the last N lines of the pod's logs?

Reply:
yes, this error has a summary of what failed during pod specialization. definitely helps to have pod's logs. but i was thinking `fission fn logs` can be useful for that.